lockbox-ios/Store/DataStore.swift

```diff
@@ -296,18 +297,38 @@ extension DataStore {
     }
 
     private func updateSyncState(from notification: Notification) {
-        let state: SyncState
-        switch notification.name {
-        case NotificationNames.FirefoxAccountVerified:
-            state = .ReadyToSync
-        case NotificationNames.ProfileDidStartSyncing:
-            state = .Syncing
-        case NotificationNames.ProfileDidFinishSyncing:
-            state = .Synced
+        Observable.combineLatest(self.storageState.take(1), self.syncState.take(1))
```

Comment:
I would move the `.take(1)` to after the `combineLatest`; I usually try to avoid nesting operators like that unless absolutely necessary.

Reply:
:+1: